packages/ember-metal/lib/utils.js

```diff
@@ -179,7 +179,8 @@ export function generateGuid(obj, prefix) {
   @return {String} the unique guid for this instance.
 */
 export function guidFor(obj) {
-  if (obj && obj[GUID_KEY]) {
+  let type = typeof obj;
+  if ((type === 'object' || type === 'function') && obj !== null && obj[GUID_KEY]) {
```

Comment:
good point, I refactored that a bit to avoid the `!== null` check for functions.